### ISSUE TITLE
Plugin metadata submenu

### ISSUE THREAD

**Issue by soulbridge** [VCVRack/Rack#254]
it would be useful to read the version of plugins under VCV software and on the list of plugins on the wiki to always know if you're update with all modules.

**Comment by odahan** [comment]
that's a good idea. Not just for information but in case of troubles it should be easy for the user to report the version of modules (and main vcv rack exe). It will help support in the future.

**Comment by soulbridge** [comment]
![cattura di schermata 48](https://user-images.githubusercontent.com/32793048/31720659-722dc39c-b417-11e7-8332-0fd09e52c509.png)
I thought that can be shown in semi-transparent aligned on the right of productor's names

**Comment by AndrewBelt** [comment]
This prompts a more general discussion about plugin versions. Should plugin versions always be based on the Rack version? How do we manage multiple releases of a plugin compatible with a single version of Rack?

**Comment by soulbridge** [comment]
as a programmer i would say: yes plugins have to be always based on the Rack version, now you have to think VCV is in development but in the future so many functions will be added less frequently and so an improvement from VCV X to VCV X+1(X.5 or whatever) won't be a problem to developers of plugins. For the second i think an outdate plugin have to be left apart. Dont know if you agree with me.

**Comment by AndrewBelt** [comment]
Okay, should version numbers of plugins be standardized in any way? Maybe something like MyPlugin v0.4? Basically v{RACK_MAJOR}.{RACK_MINOR}{LETTER} ?

**Comment by soulbridge** [comment]
yeah if you want, the first numbers can be a reference of rack version compatibility
and others just like a reference for the plugin version
and as u said: 
myplugins    v0.4 -> v0.4a -> v0.4b etc.

**Comment by AndrewBelt** [comment]
How about this: Starting with Rack v0.4.0, all Rack vX.Y.* versions will be compatible with vX.Y.* plugins. This means that Rack v0.5.1 will load a v0.5.0 plugin, and Rack v0.4.0 will load a v0.4.5 plugin. Users only need to care about matching the minor versions.

OR, we could use true semantic versioning and bump the major version by one every time the API breaks. I actually like that, despite the weirdness of a Rack v56.1 release.

**Comment by AndrewBelt** [comment]
Proposed in #266

**Comment by w3stgaard** [comment]
"think an outdate plugin have to be left apart" ... 

I would like to complete something.

We should think about a minimum of downward compatibility too. Think about the future, when you reopen old projects and nothing works anymore and you don`t have the old Rack or Plugin Version anymore, a publisher collapsed and so on ^^. Maybe think about an online based archive of old versions (Rack and Plugins) for compatibility issues too. I mean, we try to reopen 17 years !! old Logic Audio (Emagic) Arrangements under Windows via VM over here and it still works, sometimes even with NEW Plugin Versions of an old VST Instrument. Sounds weird, but :

Just some thoughts.

**Comment by AndrewBelt** [comment]
Here is why you cannot compare VCV plugins to VST/AU plugins in terms of compatibility
- VST/AU plugins have about 2-10 classes/structures in the plugin API. Rack has ~200.
- VST2 was released in 1999. Rack v0.3 was released in September.
- The latest version VST3.5 was released in 2011. The latest Rack version was released less than a month ago.
- VST/AU plugins' UI does integrate with the application. It simply launches inside a provided OS-native window, and there is no communication of the UI to/from the DAW. In Rack, it is intimately connected because of patching, context menus, dragging, key commands, and everything else that modules share and integrate into the system.
- VST/AU plugins process buffers of audio, while VCV modules process single samples at a time. This requires the audio engine to be intimately connected too, instead of hiding behind a wrapper.

Before, you say "the VCV plugin format is completely hopeless for backward compatability!", realize that this problem is effectively solved through Rack's and its plugins' open sourceness. If a plugin is good but doesn't work, someone will fix it. This will not happen with closed VST/AU plugins. This isn't called "backward compatibility", because it is not needed with open source applications. The community can simply fix these compatibility problems themselves and distribute the changes.

For commercial plugins, both by me and third-party, they will always be updated by me when I update Rack's API to a new major version (See #266 ). This is one of the services they gain by their commission fee on third-party sales.

So in summary, all plugins will break with a new major version, and backward compatibility will only work within the same major version, but the problem is not as bad as you might think due to the community effort for keeping plugins up to date.

**Comment by AndrewBelt** [comment]
Solved with 5567c5f22d1966974e16c63411826fc817a8f7d1
